### PR TITLE
Firecert-1496 Merge to Documentation Branch

### PR DESCRIPTION
**Pull request by jmeier204**
# Description
Combined PR for execution and configuration documentation.

## JIRA Ticket
[FIRECERT-1496](https://ccp.sys.comcast.net/browse/FIRECERT-1496)
[FIRECERT-1497](https://ccp.sys.comcast.net/browse/FIRECERT-1497)

## Additional Info (Test steps / Steps to reproduce etc.)
Details on how to test the feature

### CODE REVIEW

**Review by jbigel — CHANGES_REQUESTED** [review]
Documention.md
- Configuration Options: need to add a description of the section so users know if they want to learn more. The what it is and why they'd want it. Can be the same as the overview in the section.
- Test Executions: need to add a description of the section so users know if they want to learn more. The what it is and why they'd want it. Can be the same as the overview in the section.

I think I may back track on not having subsections after reviewing the document because we are essentially covering a few different things in this doc. I see three top level sections:

- Supported Modes of Execution
- Test Types
- Modes of Communication

Under Supported Modes of Execution we should have:

- UI
- Standalone
- Remotely

For each section we should
- Explain what this mode is, what makes it distinct from the others, and why you would want to chose this mode in Background.
	- For Remotely: I think a lot of this is info more needed in Setup and Usage. PubSub Client = setup, Sample Intent = Usage.
- Explain any pre-reqs needed to use this mode in Setup.
	- For UI: Current setup feels more like usage instructions. If user just needs to launch the app on the device and no other special pre-req setup is required we can say just that.
	- For standalone: Give clearer steps on how to set the url parameter - what does that look like for the user? How do they do it with an intent message. We don't need to explain how to do send intent, we can link to intent doc, but should explain what parameters need to be sent. For linking, Intent Reader doc is currently being worked on: https://github.com/rdkcentral/firebolt-certification-app/blob/FIRECERT-1500/docs/IntentReader.md.
- Explain how to trigger sanity and regression for this mode and link to specific Test Types defined in this doc.
	- For UI: can link the GUI doc for specific instructions on navigation.
	- For standalone: there is a reporting section of docs so we don't need to mention reporting here.
	- For remotely: I don't think we need to give full sample intent but can just explain params needed to trigger and link to https://github.com/rdkcentral/firebolt-certification-app/blob/FIRECERT-1500/docs/intentReaderHandlers/RunTestHandler.md for specifics on triggering. This probably applies to standalone too.

Under Test Types we should have:

- Sanity
- Regression

For each section we don't need need Setup and Usage since that applies more to the supported modes of execution. Instead we just explain what it is and what it does and maybe a little bit about how it does it.

For Modes of Communication we need a lot more information here. As a reader I am left confused after reading the current content. 
Transport: why do we bypass? How? What does it do/affect others?
I'm also wondering if this applies to executing or configuration more?

I didn't look over Configurations in too much depth but from a brief look I think we need a lot more info. You may be able to use some of the comments I gave for Executions to as guidelines improve configuration. 
A couple specific notes:
- It seems if MFOS is the only supported target I'm not sure it makes sense to have a supported targets section but can just have `Running with MFOS`.
- All the setup/usage instructions should explain what it would look like for a user to start using the feature
- For url parameters I'm pretty sure there are a lot more url params you have not included.
- Each url parameter should explain what it is and how it affects FCS runs.

File: docs/Execution.md [hunk 1]
```diff
@@ -1,92 +1,133 @@
...
 
-Before you begin using FCA with API Access, there is one essential pre-requisite:
-
-  - **PubSub Client**: In order to use the API access of FCA you first need to implement a PubSub client to handle incoming and outgoing calls. For information on setting up your own PubSub client please see the [PubSub documentation](plugins/PubSub.md).
-
-TODO: add details for how to configure FCA's PubSub client topics, listeners/subscriptions, compressions, etc...
-TODO: add details for how to configure a PubSub client to talk to FCA's PubSub Client.
+- **PubSub Client**: 
+A PubSub client is necessary to trigger FCA remotely.
+Users can also send messages via WebStock (WSS) on local area network (LAN).
```
[the remark this answers is not in the document]
Comment (ksentak) [reply docs/Execution.md:44]: The line via WebStock

File: docs/Execution.md [hunk 2]
```diff
@@ -1,92 +1,133 @@
...
+
+## Regression
+
+Regression testing involves in-depth testing to ensure new features or changes haven't adversely affected existing functionality.
+Create custom test cases with specific parameters tailored to the features or changes being tested.
+Send requests with custom parameters and verify responses against predefined data sets for targeted testing.
+to trigger those tests, build a framework communicating with your device, and send requests and verify responses as needed.
+
+
+# Modes of communication
```
[the remark this answers is not in the document]
Comment (ksentak) [reply docs/Execution.md:126]: Capitalize Communication

File: docs/Execution.md [hunk 3]
```diff
@@ -1,92 +1,133 @@
...
+Sanity testing ensures that all APIs function correctly at a basic level, serving as a quick check method.
+Execute the Sanity Test Suite from the UI, as described in the 'Supported Ways of Execution' section.
+This suite tests all APIs by invoking predefined calls under each API, offering a comprehensive overview of basic functionality.
+
+## Regression
+
+Regression testing involves in-depth testing to ensure new features or changes haven't adversely affected existing functionality.
+Create custom test cases with specific parameters tailored to the features or changes being tested.
+Send requests with custom parameters and verify responses against predefined data sets for targeted testing.
+to trigger those tests, build a framework communicating with your device, and send requests and verify responses as needed.
```
Comment (ksentak) [inline docs/Execution.md:123]: For regression testing a user may want to send requests with custom parameters and verify responses against predefined data sets for targeted testing. To trigger those tests, users may be inclined to build out a framework that allows for communicating with your device, and can both send requests and verify responses as needed.

**Review by ksentak — CHANGES_REQUESTED** [review]
Lot of comments are spell checks and formatting. Once Execution changes are made that should be good to go. Configurations doc needs some more work.

File: docs/Execution.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (omarsabha91) [reply docs/Execution.md:126]: @ksentak for this one Modes of Communication is introduced as h1 in the TOC. should we change that ?

[the remark this answers is not in the document]
Comment (jbigel) [reply docs/Execution.md:126]: Seconded, it should be 
```
# Title
## TOC
## Section
```

Since we have sub sections aside from `Background`/`Setup`/`Usage` I'm thinking maybe we just make them bold because h4 looks off. So like:
```
# FCA: Executing Tests
## Table of Contents
## Supported ways of Execution
### UI
**Background**:
**Setup**:
**Usage**:
etc...
```

File: docs/Execution.md [hunk 4]
```diff
@@ -1,92 +1,133 @@
...
+
+
+# Modes of communication
+
+Different modes of communication offer flexibility in test approach, tailored to specific testing needs.
```
Comment (jbigel) [inline docs/Execution.md:128]: Maybe we can provide a couple use cases to give context and highlight what we mean by `flexibility in test approach, tailored to specific testing needs.`

File: docs/Configurations.md [hunk 5]
```diff
@@ -1,99 +1,86 @@
...
-## Supported targets
-
-EXISTING CONTENTS(needs revision):
-> - MFOS - `https://github.com/rdkcentral/mock-firebolt`
+## Running with MFOS
 
 ### Background
 
-TODO: add details
+Mock Firebolt (MFOS) is a simulated environment that FCA can interact with.
```
Comment (jbigel) [inline docs/Configurations.md:17]: Think we can add more details to explain to the user, what is MFOS simulating? The device the firmware, the APIs?

Comment (jbigel) [reply docs/Configurations.md:17]: Very abbreviated description of what MFOS is and does would be good and we can link to MFOS documentation guide for more info.

File: docs/Configurations.md [hunk 6]
```diff
@@ -1,99 +1,86 @@
...
 
 ### Setup
 
-TODO: add details
+- *MFOS*: Accessible at [Mock-Firebolt](https://github.com/rdkcentral/mock-firebolt)
```
Comment (jbigel) [inline docs/Configurations.md:21]: Can make this a full sentence, Like:
The open source GitHub project is accessible at...

File: docs/Configurations.md [hunk 7]
```diff
@@ -1,99 +1,86 @@
...
+- Use Mock Firebolt's [usage documentation](https://github.com/rdkcentral/mock-firebolt#usage-local) to get set up. 
 
 ### Usage
 
-TODO: add details
+Utilize MFOS for testing in a controlled, simulated environment. MFOS can be ideal for early-stage development and testing.
```
Comment (jbigel) [inline docs/Configurations.md:26]: This is not sufficient instructions for users to know how to use MFOS with FCA.

File: docs/Configurations.md [hunk 8]
```diff
@@ -1,99 +1,86 @@
...
 
 ## Connect to Mock Firebolt
```
Comment (jbigel) [inline docs/Configurations.md:28]: We don't need two sections, `Running with MFOS` and `Connect to Mock Firebolt`, they should be combined into one.

File: docs/Configurations.md [hunk 9]
```diff
@@ -1,99 +1,86 @@
...
-  - If FCA systemui=true, FCA acts as the base app in case of ripple. The background color will be changed to purple, and it will display one more button as "Launch FCA app" to launch FCA as third-party app on Ripple devices.
-- TestContext: testContext=true
-  - If testContext=true, it will add the field context in mocha report generated
-```
+
 
 ### Background
 
-TODO: add details
+URL parameters allow customization of the FCA's behavior, enabling testers to tailor the testing environemt to specific needs.
```
Comment (jbigel) [inline docs/Configurations.md:50]: A couple example use cases would be good here.

File: docs/Configurations.md [hunk 10]
```diff
@@ -1,99 +1,86 @@
...
 
 ### Setup
 
-TODO: add details
+- Use various parameters like 'platform', 'lifecycle_validation','mf','systemui' and more as per the requirment.
```
Comment (jbigel) [inline docs/Configurations.md:54]: How do you set up FCA with these parameters.
Also maybe add something like:
See ### Usage for specific use cases.

File: docs/Configurations.md [hunk 11]
```diff
@@ -1,99 +1,86 @@
...
+- These parameters modify how the FCA interacts with the environemtn and the test suite.
```
Comment (jbigel) [inline docs/Configurations.md:55]: Not sure this is necessary, if we explain in Usage, but to me doesn't really read like a setup instruction.

File: docs/Configurations.md [hunk 12]
```diff
@@ -1,99 +1,86 @@
...
 
 ### Usage
 
-TODO: add details
+Customize the test execution and reporting based on the provided URL parameters for targeted testing scenarios.
+
+**Parameters usage:**
```
Comment (jbigel) [inline docs/Configurations.md:61]: Can this be a table with columns
Parameter | Description | Possible Values | and maybe Notes?

File: docs/Configurations.md [hunk 13]
```diff
@@ -1,99 +1,86 @@
...
+
 
 ## App Types
- 
-TODO: add details
 
 ### Background
 
-TODO: add details
+Different app types in FCA denote various categories or classifications of applications that can be tested using the platform.
```
Comment (jbigel) [inline docs/Configurations.md:77]: Can we add some examples/use cases?

File: docs/Configurations.md [hunk 14]
```diff
@@ -1,99 +1,86 @@
...
 
 ### Background
 
-TODO: add details
+Different app types in FCA denote various categories or classifications of applications that can be tested using the platform. 
 
 ### Setup
 
-TODO: add details
+- Define the app types as part of the configuration before initiating tests.
```
Comment (jbigel) [inline docs/Configurations.md:81]: How?

File: docs/Configurations.md [hunk 15]
```diff
@@ -1,99 +1,86 @@
...
+- This may involve specifying the type of applocation under test, such as media apps, utility apps, etc.
```
Comment (jbigel) [inline docs/Configurations.md:82]: Examples/use cases?

File: docs/Configurations.md [hunk 16]
```diff
@@ -1,99 +1,86 @@
...
 
 ### Usage
 
-TODO: add details
+- Select the appropriate app type to ensure that tests are relevant and effective for the specific category of the application.
```
Comment (jbigel) [inline docs/Configurations.md:86]: This doesn't tell the user how to use app types in FCA.

**Review by jbigel — CHANGES_REQUESTED** [review]

File: docs/Execution.md [hunk 17]
```diff
@@ -1,92 +1,133 @@
...
 
 ### Setup
 
-Before you begin using FCA with API Access, there is one essential pre-requisite:
-
-  - **PubSub Client**: In order to use the API access of FCA you first need to implement a PubSub client to handle incoming and outgoing calls. For information on setting up your own PubSub client please see the [PubSub documentation](plugins/PubSub.md).
-
-TODO: add details for how to configure FCA's PubSub client topics, listeners/subscriptions, compressions, etc...
-TODO: add details for how to configure a PubSub client to talk to FCA's PubSub Client.
+- **PubSub Client**:
```
[the remark this answers is not in the document]
Comment (omarsabha91) [reply docs/Execution.md:44]: this comment should be marked outdated, not sure why it's not. the change for this should be in.

File: docs/Execution.md (hunk 4 above)
Comment (omarsabha91) [reply docs/Execution.md:127]: @parag-pv do we have anything in ThorTV that can manifest the difference between communication via the SDK or directly with the device ?
only use cases we have is suits, not sure how we can give an example here.

File: docs/Configurations.md (hunk 5 above)
Comment (omarsabha91) [reply docs/Configurations.md]: @jbigel  link to documentation is linked in the setup, should I move it from there ?

File: docs/Configurations.md (hunk 9 above)
Comment (omarsabha91) [reply docs/Configurations.md:50]: @jbigel is this still needed after adding all the details into a table ?

File: docs/Configurations.md (hunk 16 above)
Comment (omarsabha91) [reply docs/Configurations.md:84]: @jbigel in the setup, it's mentioned now that App Type can be sent via the intent message. 
any other addition for this section ?

File: docs/Execution.md (hunk 4 above)
Comment (parag-pv) [reply docs/Execution.md]: You could mention test scenarios where the API call should not go through the SDK and scenarios that should. When in transport mode, the SDK is bypassed which would be desirable in some use cases. Example would be a case where SDK would reject a requests based on the validations that it imposes.

File: docs/Configurations.md [hunk 18]
```diff
@@ -1,99 +1,88 @@
 # FCA: Configurations
 
-TODO: add intro
+This document gives the various configurations available for Firebolt Certification App ( FCA). It provides insights into setting up and utiizing different environemtns, parameters, and app types to facilitate effective testing and certification of applications using firebolt technology.
```
[the remark this answers is not in the document]
Comment (omarsabha91) [reply docs/Configurations.md]: Done, removed extra space and corrected the spelling of environment.

File: docs/Configurations.md [hunk 19]
```diff
@@ -1,99 +1,88 @@
...
-## Supported targets
-
-EXISTING CONTENTS(needs revision):
-> - MFOS - `https://github.com/rdkcentral/mock-firebolt`
+## Running with MFOS
 
 ### Background
 
-TODO: add details
+Mock Firebolt OS (MFOS) is a simulated environment that FCA can interact with.
```
[the remark this answers is not in the document]
Comment (omarsabha91) [reply docs/Configurations.md]: done, removed OS.

File: docs/Configurations.md [hunk 20]
```diff
@@ -1,99 +1,88 @@
...
 
 ### Setup
 
-TODO: add details
+- *MFOS*: Accessible at `https://github.com/rdkcentral/mock-firebolt`
+- Follow the instructions provided in the repository to setup the MFOS environment
```
[the remark this answers is not in the document]
Comment (omarsabha91) [reply docs/Configurations.md]: fixed
- Use Mock Firebolt's [usage documentation](https://github.com/rdkcentral/mock-firebolt#usage-local) to get set up.

File: docs/Documentation.md [hunk 21]
```diff
@@ -24,11 +24,20 @@ When you launch the Firebolt Certification Application (FCA) GUI, you gain acces
 
 ## Configuration Options
 
-To learn more about configuration options see the [configuration documentation](./Configurations.md).
+This document gives the various configurations available for Firebolt Certification App ( FCA). It provides insights into setting up and utiizing different environemtns, parameters, and app types to facilitate effective testing and certification of applications using firebolt technology.
```
[the remark this answers is not in the document]
Comment (omarsabha91) [reply docs/Documentation.md]: fixed typo and space

File: docs/Execution.md [hunk 22]
```diff
@@ -1,92 +1,133 @@
...
-#### Regression Test Execution
+  - [UI](#ui)
+  - [Remotely](#remotely)
+- [Test Types](#test-types)
+  - [Sanity](#sanity)
+  - [Regression](#regression)
+- [Modes of Communication](#modes-of-communication)
 
-TODO: add details
+# Supported ways of Execution
```
[the remark this answers is not in the document]
Comment (omarsabha91) [reply docs/Execution.md]: capitalized 
## Supported Ways of Execution

File: docs/Execution.md [hunk 23]
```diff
@@ -1,92 +1,133 @@
...
-TODO: add details
-
-
-## Executing Remotely
+## Remotely
 
 ### Background
 
-FCA extends its capabilities by offering remote accessibility through API access, presenting automation opportunities for users responsible for managing test suites, as well as those without direct device access. By utilizing a PubSub client, users can dispatch requests to integrated devices running FCA, enabling the initiation of various tests, Firebolt calls, and other tasks (see [PubSub Handlers documentation](pubSubHandlers/PubSubHandlers.md)). However, it's essential to note that implementing a PubSub client is a prerequisite (refer to the [PubSub documentation](plugins/PubSub.md) for guidance). This remote communication feature provides distinct advantages, including the ability to schedule automated test suites, reduce the dependency on physical devices for developers, and more.
+FCA extends its capabilities by offering remote accessibility through API access,benefiting users managing test suites or without direct device access.
```
[the remark this answers is not in the document]
Comment (omarsabha91) [reply docs/Execution.md]: space added after comma

File: docs/Execution.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (omarsabha91) [reply docs/Execution.md]: updated to 
- **PubSub Client**: 

A PubSub client is necessary to trigger FCA remotely.
Before using FCA with API access, implement a PubSub client as per the [PubSub documentation](plugins/PubSub.md).

File: docs/Execution.md [hunk 24]
```diff
@@ -1,92 +1,133 @@
...
+            "context": {
+                "source": "device"
+            }
+        }
+    }
+}
+
+```
+
+for more specific triggering follow instructions
```
[the remark this answers is not in the document]
Comment (omarsabha91) [reply docs/Execution.md]: Done

File: docs/Execution.md [hunk 25]
```diff
@@ -1,92 +1,133 @@
...
+for more specific triggering follow instructions 
+[RunTest Handler documentation](intentReaderHandlers/RunTestHandler.md).
+
+
+# Test Types
+
+## Sanity
+
+Sanity testing ensures that all APIs function correctly at a basic level, serving as a quick check method.
+Execute the Sanity Test Suite from the UI, as described in the 'Supported Ways of Execution' section.
```
[the remark this answers is not in the document]
Comment (omarsabha91) [reply docs/Execution.md]: updated to 
### Sanity

Sanity testing ensures that all APIs function correctly at a basic level, serving as a quick check method.
Execute the Sanity Test Suite, as described in the [Supported Ways of Execution](#supported-ways-of-execution) section.
This suite tests all APIs by invoking predefined calls under each API, offering a comprehensive overview of basic functionality.

File: docs/Execution.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (omarsabha91) [reply docs/Execution.md]: updated to 
## Modes of communication
also updated all the other titles in the file according to 
# FCA: Executing Tests
## Table of Contents
## Supported ways of Execution
### UI
**Background**:
**Setup**:
**Usage**:
etc...

File: docs/Documentation.md [hunk 26]
```diff
@@ -24,11 +24,16 @@ When you launch the Firebolt Certification Application (FCA) GUI, you gain acces
...
```
[the remark this answers is not in the document]
Comment (omarsabha91) [reply docs/Documentation.md]: updated description to this :
There are various configurations available for Firebolt Certification App (FCA), Including setting up and utilizing different environments, parameters, and app types to facilitate effective testing and certification of applications using firebolt technology.
[configuration documentation](./Configurations.md).

File: docs/Execution.md [hunk 27]
```diff
@@ -1,92 +1,133 @@
...
-TODO: add details
+Manual execution via the user interface allows for interactive and immediate testing, giving testers direct control over the test process.
 
 ### Setup
 
-TODO: add details
+Launch the app on the device.
 
 ### Usage
+Navigate through the Firebolt UI.
```
[the remark this answers is not in the document]
Comment (omarsabha91) [reply docs/Execution.md]: updated to this:

**Usage**:
- Navigate through the Firebolt UI.
- Choose the communication mode: SDK or Transport.
- Select the desired SDK mode: CoreSDK, ManageSDK, or All SDKs.
- Invoke the sanity suite or specific tests manually.

File: docs/Execution.md (hunk 4 above)
Comment (omarsabha91) [reply docs/Execution.md]: updated descriptions to this :
Different modes of communication offer flexibility in test approach, tailored to specific testing needs, for example invoking a device.version request would return SDK version if ran via SDK communication mode, and it'll be skipped if invoked in communication mode.

File: docs/Configurations.md (hunk 5 above)
Comment (omarsabha91) [reply docs/Configurations.md]: updated to mention what a simulated environment means in terms of MFOS :
Mock Firebolt (MFOS) is a simulated environment representing (real Firebolt running on device) that FCA can interact with.
also updated usage to :
Utilize MFOS for testing in a controlled, simulated environment. MFOS can be ideal for early-stage development and testing.
Use MFOS if you don't have a device with firebolt running on it, or don't have a device at all.

File: docs/Configurations.md (hunk 7 above)
Comment (omarsabha91) [reply docs/Configurations.md:26]: in the setup there is link to all MFOS documentation where setup info is all documented. 
also updated usage to this :
Utilize MFOS for testing in a controlled, simulated environment. MFOS can be ideal for early-stage development and testing.
Use MFOS if you don't have a device with firebolt running on it, or don't have a device at all.

File: docs/Configurations.md (hunk 8 above)
Comment (omarsabha91) [reply docs/Configurations.md]: merged both 
currently looking like this :
in TOC:
## Table of Contents

- [Running with MFOS](#running-with-mfos)
  - [How to Connect with Mock Firebolt?](#connect-to-mock-firebolt)
  
  
  
  ## Running with MFOS
  ### Connect to Mock Firebolt

File: docs/Configurations.md (hunk 10 above)
Comment (omarsabha91) [reply docs/Configurations.md]: covered in the parameters table under description column 
each of the descriptions explains the use of the parameter. 
please advise if there is specific format we can add

File: docs/Configurations.md (hunk 11 above)
Comment (omarsabha91) [reply docs/Configurations.md]: removed the second line

File: docs/Configurations.md (hunk 16 above)
Comment (omarsabha91) [reply docs/Configurations.md:84]: - Define the app types as part of the configuration before initiating tests (AppType can be sent via the intent message) [Remote Execution](https://github.com/rdkcentral/firebolt-certification-app/blob/docs/Execution.md#remotely).

File: docs/Execution.md [hunk 28]
```diff
@@ -1,92 +1,133 @@
...
+
+# Test Types
+
+## Sanity
+
+Sanity testing ensures that all APIs function correctly at a basic level, serving as a quick check method.
+Execute the Sanity Test Suite from the UI, as described in the 'Supported Ways of Execution' section.
+This suite tests all APIs by invoking predefined calls under each API, offering a comprehensive overview of basic functionality.
+
+## Regression
```
[the remark this answers is not in the document]
Comment (jbigel) [reply docs/Execution.md]: @omarsabha91 after discussing with Avani let's remove regression from this for now. I think we'd need more info and without being able to discuss how it works with TTT it would likely be more confusing than helpful to readers.

File: docs/Configurations.md (hunk 5 above)
Comment (jbigel) [reply docs/Configurations.md]: `representing` should be inside the `(`

File: docs/Configurations.md (hunk 8 above)
Comment (jbigel) [reply docs/Configurations.md]: Just one section for MFOS -`How to Connect with Mock Firebolt?` is just setup/usage instructions for the MFOS section.

File: docs/Configurations.md (hunk 9 above)
Comment (jbigel) [reply docs/Configurations.md:51]: I mean an example in the background of what a url parameter would do for you. Given the example you added how does that benefit the users.

Comment (jbigel) [reply docs/Configurations.md:51]: Like an example of why you would use a URL param.

Comment (jbigel) [reply docs/Configurations.md:51]: The other thing I'm wondering is: can you pass multiple URL prams? Can you add that detail in the Setup

File: docs/Configurations.md (hunk 10 above)
Comment (jbigel) [reply docs/Configurations.md]: Yeah I think the table you made actually covers this pretty well so can probably just remove it from setup.

File: docs/Configurations.md (hunk 12 above)
Comment (jbigel) [reply docs/Configurations.md]: This looks great. Just a few notes:
- `platform` I'm not understanding what this would be, what are values we use?
- mf: `affects how APIs in FCA respond.` using MF means, based on the configuration, it calls method on MF instead of the device right? Think we can say that more clearly.
- systemui: changing the color is just a visual queue for users right? How does it actually change the FCA behavior?
- reportingId: does it append it or replace the uuid with the passed one?
- standalone: Let's link to the standalone doc in intentReader for more usage instructions (not merged into the main doc branch yet but can still create the hyperlink): https://github.com/rdkcentral/firebolt-certification-app/blob/FIRECERT-1500/docs/IntentReader.md#standalone

Wondering if some of these params require more info on how to use them. Not necessarily all of them, for example `voiceGuidance` is pretty self explanatory, but `testContext` I'm wondering what is the additional context provided in the report? Is that based on a config or user input? Or is it a static thing added when true?
Basically wondering if you turn on any of these url params is there additional steps that need to be taken by users?

File: docs/Configurations.md (hunk 16 above)
Comment (jbigel) [reply docs/Configurations.md:85]: How do you define app types `as part of the configuration`. This is the configuration doc so we need to be more explicit on the hows.

File: docs/Configurations.md [hunk 29]
```diff
@@ -51,35 +52,33 @@ URL parameters allow customization of the FCA's behavior, enabling testers to ta
...
 ### Background
 
-Different app types in FCA denote various categories or classifications of applications that can be tested using the platform. 
+Different app types in FCA denote various SDK's of applications that can be tested using the platform. 
 
 ### Setup
 
-- Define the app types as part of the configuration before initiating tests.
-- This may involve specifying the type of applocation under test, such as media apps, utility apps, etc.
+- Define the app types as part of the configuration before initiating tests (AppType can be sent via the intent message) [Remote Execution](https://github.com/rdkcentral/firebolt-certification-app/blob/docs/Execution.md#remotely).
```
Comment (jbigel) [inline docs/Configurations.md:81]: Use a relative link when hyperlinking docs in the same repo. i.e. `[Remote Execution](./Execution.md#remotely)`

File: docs/Execution.md (hunk 4 above)
Comment (jbigel) [reply docs/Execution.md]: So if invoked in Transport do we not get the `device.version`? Or does it get the actual version of the device?

File: docs/Configurations.md (hunk 7 above)
Comment (jbigel) [reply docs/Configurations.md]: Right but that is setup instructions. Can we add an usage instructions on how to use MFOS with FCA. Think of an example of how we use MFOS to enhance testing.

File: docs/Configurations.md (hunk 8 above)
Comment (jbigel) [reply docs/Configurations.md:12]: Looks much better. A few things
- Comment from above on step by step instructions on how to use FCA with MFOS.
- Change the TOC link to match the section title.
- Think `Use MFOS if you don't have a device with firebolt running on it, or don't have a device at all.` should go in background.

File: docs/Configurations.md (hunk 9 above)
Comment (jbigel) [reply docs/Configurations.md:38]: Looks good but can the examples and benefits section go above Setup?

File: docs/Configurations.md (hunk 12 above)
Comment (jbigel) [reply docs/Configurations.md]: Looks good, just two small notes:
- Seems like the `Description` and `Notes` for system ui feels like we repeat info between the two columns.
- In reportingID `Description` need a space after URL.

File: docs/Configurations.md [hunk 30]
```diff
@@ -5,42 +5,29 @@ This document gives the various configurations available for Firebolt Certificat
...
-
-**Background**
-
-Connecting to Mock Firebolt enables testing against a simulated Firebolt OS, providing a safe and flexible enviroment for development and testing.
-
-**Setup**
-
 - Clone the Repository [Mock-Firebolt](https://github.com/rdkcentral/mock-firebolt)
 - Follow the instructions provided in the repository to start the Mock Firebolt.
 - Launch the Firebolt Cretification App and add `mf=true` to the app launch URL to connect to the mock environment.
```
Comment (jbigel) [inline docs/Configurations.md:25]: Can we link to the url params section here?

File: docs/Configurations.md [hunk 31]
```diff
@@ -52,23 +39,37 @@ URL parameters allow customization of the FCA's behavior, enabling testers to ta
...
+| **reportingId**      | Appends this ID to the report name to form the reporting URL(it replaces the uuid with the passed value).                                                                                                                | (custom ID)               | Customize the report name for easier identification and tracking.                                                    |
+| **standalone**       | Designed for automated testing scenarios, enabling detailed reports and logs within FCA independently of other apps or frameworks. Set to true for standalone execution.     | true / false              | [Standalone execution guide](https://github.com/rdkcentral/firebolt-certification-app/blob/FIRECERT-1500/docs/IntentReader.md#standalone) Essential for tests requiring independence from other apps or frameworks, with detailed reporting and logging needs.|
```
Comment (jbigel) [inline docs/Configurations.md:72]: Use a relative hyperlink i.e `./IntentReader.md#standalone`

**Review by jbigel — CHANGES_REQUESTED** [review]

File: docs/Configurations.md (hunk 12 above)
Comment (omarsabha91) [reply docs/Configurations.md]: done

File: docs/Configurations.md (hunk 31 above)
Comment (omarsabha91) [reply docs/Configurations.md]: done

File: docs/Execution.md (hunk 4 above)
Comment (omarsabha91) [reply docs/Execution.md]: it get's skipped, we don't get the device version.

File: docs/Configurations.md (hunk 7 above)
Comment (omarsabha91) [reply docs/Configurations.md]: on FCA side only the url param is required, that's the only difference to run it against MFOS.
using MFOS with FCA has good documentation in [MFOS](https://github.com/rdkcentral/mock-firebolt?tab=readme-ov-file#usage-local)

File: docs/Configurations.md (hunk 8 above)
Comment (omarsabha91) [reply docs/Configurations.md:12]: updated to this:
## Connect to Mock Firebolt

### Background

Mock Firebolt (MFOS) is a simulated environment (representing real Firebolt running on device) that FCA can interact with.
Connecting to Mock Firebolt enables testing against a simulated Firebolt OS, providing a safe and flexible enviroment for development and testing.
Use MFOS if you don't have a device with firebolt running on it, or don't have a device at all.

### Setup

- *MFOS*: The open source GitHub project is accessible at [Mock-Firebolt](https://github.com/rdkcentral/mock-firebolt)
- Use Mock Firebolt's [usage documentation](https://github.com/rdkcentral/mock-firebolt#usage-local) to get set up. 
- Clone the Repository [Mock-Firebolt](https://github.com/rdkcentral/mock-firebolt)
- Follow the instructions provided in the repository to start the Mock Firebolt.
- Launch the Firebolt Certification App and add `mf=true` to the app launch URL to connect to the mock environment, visit the [URL Parameters](#supported-url-parameters) for details on sending Parameters.

### Usage

Utilize MFOS for testing API integrations and app behavior in a controlled, simulated environment. MFOS can be ideal for early-stage development and testing.
On FCA side only passing `mf=true` URL parameter is required. For detailed instructions on the setup and usage check [MFOS Usage](https://github.com/rdkcentral/mock-firebolt?tab=readme-ov-file#usage-local).

File: docs/Configurations.md (hunk 9 above)
Comment (omarsabha91) [reply docs/Configurations.md:38]: done

**Review by omarsabha91 — APPROVED** [review]

File: docs/Execution.md (hunk 4 above)
Comment (parag-pv) [reply docs/Execution.md]: @omarsabha91, rather than mentioning `skipped`, which might be confusing, could you rephrase it to convey that the device.version response in transport mode will not contain the `sdk` field in response but SDK mode will? 

The current doc seems to covey that transport mode will not give a response for device.version

Comment (omarsabha91) [reply docs/Execution.md]: updated the description to:
Different modes of communication offer flexibility in test approach, tailored to specific testing needs, for example invoking a device.version request would return SDK version if ran via SDK communication mode, whereas in Transport mode the response will not contain `sdk` field.

File: docs/Configurations.md (hunk 7 above)
Comment (jbigel) [reply docs/Configurations.md]: So asking a question here, we have our normal nightly pipelines and we have a mfos pipeline - what is the difference in the purpose they serve? How do they differ?
I see here: https://github.comcast.com/lightning-automation/thorTV-Tests/blob/development/ci/tasks/mfos_certification.sh#L253-L259 we do upload specific files? Why/what is that for?

cc// @parag-pv

File: docs/Configurations.md (hunk 13 above)
Comment (jbigel) [reply docs/Configurations.md]: Seeing how the other background sections are not bulleted can make that consistent here.

File: docs/Configurations.md (hunk 7 above)
Comment (parag-pv) [reply docs/Configurations.md]: For MFOS, the yaml/json files are uploaded to override method responses. By default MF would return the mock values, but for any method that requires a custom value to be returned (for validation in TTT), we upload the overrides from MFOS call flow repo https://github.comcast.com/lightning-automation/MFOS_Call_Flow

This is more of a TTT/suite thing than FCA, we wouldn't need to mention that for FCA documentation since it's nothing specific to FCA.

File: docs/Configurations.md (hunk 13 above)
Comment (omarsabha91) [reply docs/Configurations.md]: done

**Review by omarsabha91 — APPROVED** [review]

File: docs/Configurations.md (hunk 7 above)
Comment (jbigel) [reply docs/Configurations.md]: okay thanks for the info.

**Review by jbigel — APPROVED** [review]
